dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/sms/TrackedEntityRegistrationSMSListener.java

```diff
@@ -142,16 +142,17 @@ protected void postProcess(
               patientAttributeValues.add(trackedEntityAttributeValue);
             });
 
-    long trackedEntityId = 0;
     if (!patientAttributeValues.isEmpty()) {
-      trackedEntityId =
-          trackedEntityService.createTrackedEntity(trackedEntity, patientAttributeValues);
+      trackedEntityService.createTrackedEntity(trackedEntity, patientAttributeValues);
     } else {
       sendFeedback("No TrackedEntityAttribute found", senderPhoneNumber, WARNING);
     }
 
     enrollTrackedEntity(
-        trackedEntityService.getTrackedEntity(trackedEntityId), program, orgUnit, occurredDate);
+        trackedEntityService.getTrackedEntity(trackedEntity.getUid()),
```

Comment:
In the PR description you say

> getTrackedEntity(String uid) was used instead as it is auditing the read.

do you mean you are using `getTrackedEntity(String uid)` instead of `getTrackedEntity(long)`?

Reply:
Yes. We will eventually remove also the other method, but we need to discuss about the read auditing part and I didn't want to do it just yet.